src/main/scala/beam/utils/scenario/BeamScenarioLoader.scala

```diff
@@ -176,50 +177,71 @@ class BeamScenarioLoader(
+            val person = population.getPersons.get(Id.createPersonId(personId.id))
+
+            val currentPlan = PopulationUtils.createPlan(person)
+            person.addPlan(currentPlan)
+
+            val personWithoutSelectedPlan = person.getSelectedPlan == null
+            val isCurrentPlanIndexSelected = listOfElementsGroupedByPlan.head.planSelected
+            val isLastPlanIteration = person.getPlans.size() == listOfElementsGroupedByPerson.size
+            if (personWithoutSelectedPlan && (isCurrentPlanIndexSelected || isLastPlanIteration)) {
+              person.setSelectedPlan(currentPlan)
+            }
+
+            listOfElementsGroupedByPlan.foreach { planElement =>
+              if (planElement.planElementType.equalsIgnoreCase("leg")) {
+                buildAndAddLegToPlan(currentPlan, planElement)
+              } else if (planElement.planElementType.equalsIgnoreCase("activity")) {
+                buildAndAddActivityToPlan(currentPlan, planElement)
+              }
+            }
         }
-      }
     }
     population
   }
 
-  private def selectedPlanFromPersonOrBuildNew(person: Person): Plan = {
-    Option(person.getSelectedPlan).getOrElse {
-      val plan = PopulationUtils.createPlan(person)
-      person.addPlan(plan)
-      person.setSelectedPlan(plan)
-      plan
+  private def buildAndAddActivityToPlan(currentPlan: Plan, planElement: PlanElement): Activity = {
+    assertActivityHasLocation(planElement)
+    val coord = if (beamScenario.beamConfig.beam.exchange.scenario.convertWgs2Utm) {
+      geo.wgs2Utm(new Coord(planElement.activityLocationX.get, planElement.activityLocationY.get))
+    } else {
+      new Coord(planElement.activityLocationX.get, planElement.activityLocationY.get)
     }
+    val activityType = planElement.activityType.getOrElse(
+      throw new IllegalStateException(
+        s"planElement is `activity`, but `activityType` is None. planInfo: $planElement"
+      )
+    )
+    val act = PopulationUtils.createAndAddActivityFromCoord(currentPlan, activityType, coord)
+    planElement.activityEndTime.foreach { endTime =>
+      act.setEndTime(endTime)
+    }
+    act
+  }
+
+  private def buildAndAddLegToPlan(currentPlan: Plan, planElement: PlanElement): Leg = {
+    val leg = PopulationUtils.createAndAddLeg(currentPlan, planElement.legMode.getOrElse(""))
+    planElement.legDepartureTime.foreach(v => leg.setDepartureTime(v.toDouble))
+    planElement.legTravelTime.foreach(v => leg.setTravelTime(v.toDouble))
+
+    val legRoute = leg.getRoute
+    planElement.legRouteDistance.foreach(legRoute.setDistance)
+    planElement.legRouteStartLink.foreach(v => legRoute.setStartLinkId(Id.create(v, classOf[Link])))
+    planElement.legRouteEndLink.foreach(v => legRoute.setEndLinkId(Id.create(v, classOf[Link])))
+    planElement.legRouteTravelTime.foreach(v => legRoute.setTravelTime(v))
+    // TODO: could not find way to setup planElement.legRouteLinks and planElement.legRouteType
```

Comment:
Are you about how to add the route to the leg? Please, look to the example https://github.com/LBNL-UCB-STI/beam/blob/develop/src/main/java/beam/physsim/jdeqsim/AgentSimToPhysSimPlanConverter.java#L423